src/background/services/sendToPort.ts

```diff
@@ -1,24 +1,22 @@
 import type { Runtime } from 'webextension-polyfill';
-import {
-  BACKGROUND_TO_POPUP_CONNECTION_NAME as CONNECTION_NAME,
-  type BackgroundToPopupMessage,
-  type BackgroundToPopupMessagesMap,
-} from '@/shared/messages';
 import type { Cradle } from '@/background/container';
+import type { BackgroundToPortMessagesMap } from '@/shared/messages';
 
-export class SendToPopup {
+export class SendToPort<Message extends BackgroundToPortMessagesMap> {
   private browser: Cradle['browser'];
-
-  private isConnected = false;
   private port: Runtime.Port;
+  private isConnected = false;
 
-  constructor({ browser }: Cradle) {
+  constructor(
+    { browser }: Cradle,
+    private readonly connectionName: string,
```

Comment:
I think we'd need to extend Cradle here.
```suggestion
    { browser, connectionName }: Cradle & { connectionName: string },
```

Reply:
agree! if we gonna use `inject`